src/System.Text.Json/tests/JsonDateTimeTestData.cs

```diff
@@ -229,6 +229,9 @@ public static IEnumerable<object[]> InvalidISO8601Tests()
             yield return new object[] { "\"199\\u0037-07\\u002d16T1\\u0039:20:30.4555555+\\u002dZ\"" };
             // Proper format but invalid calendar date, time, or time zone designator fields 1997-00-16
             yield return new object[] { "\"\\u0031\\u0039\\u0039\\u0037\\u002d\\u0030\\u0030\\u002d\\u0031\\u0036\"" };
+
+            // High byte expansion
+            yield return new object[] { "\"" + new string('\u20AC', JsonConstants.MaximumEscapedDateTimeOffsetParseLength) + "\"" };
```

Comment:
I think this test case would just bail early (in the `IsValidDateTimeOffsetParseLength` check) since 254 > 252.

Shouldn't you use `JsonConstants.MaximumEscapedDateTimeOffsetParseLength - 2` to account for the surrounding quotes?

Reply:
`IsValidDateTimeOffsetParseLength` checks `MaximumEscapedDateTimeOffsetParseLength` inclusive, so I think it should be valid? 🤔 It did at least fail with the previous code (not accounting for the expansion).

Reply:
Aaah, of course. The quotes! 🙈

Reply:
> some other tests failed

Right. A single token JSON string in a payload (when parsing using JsonDocument, etc.) would fail since `\u20AC\u20AC\u20AC...` is invalid JSON. `"\u20AC\u20AC\u20AC..."` (within quotes) is valid :)